src/esm_parser/esm_parser.py

```diff
@@ -488,18 +488,39 @@ def attach_single_config(config, path, attach_value, all_config=None, **kwargs):
     else:
         print("Could not find ", path + "/" + attach_value)
         sys.exit(1)
-    # DB this is a try:
-    dict_merge(config, attachable_config, **kwargs)
+    deep_update_further_reading(config, attachable_config) #, **kwargs)
     # config.update(attachable_config)
 
 
+def deep_update_further_reading(config, further_reading_config):
+    for key, value in further_reading_config.items():
+        if isinstance(value, dict) and isinstance(config.get(key, None), dict) and config.get(key, None) is not None:
+            deep_update_further_reading(config[key], value)
+        elif key in config and value != config[key]:
+            raise ValueError(
+                f"Key {key} already exists in config and has a different value."
+            )
+        elif isinstance(value, list):
+            if isinstance(config.get(key, None), list):
+                config[key] += value
+            elif not isinstance(config.get(key, None), list) and config.get(key, None) is not None:
+                raise TypeError("meep meep, the roadrunner wins, silly cayote")
+            else:
+                config[key] = value
+
+        else:
+            config[key] = value
+
+
```

Comment:
Can we double check if this change is needed or if the old behavior works adequately? Maybe we now have two functions that effectively do the same job

Reply:
I am merging this branch into the branch `feat/remove_environment_changes` and making this comment be a TODO in the description